src/main/java/homework/HomeworkGetter.java

```diff
@@ -33,8 +33,7 @@ public HomeworkGetter(LoginCredentials creds) {
 			final HtmlPage page2 = button.click();
 			if (page2.getUrl() == page.getUrl())
 				throw new WrongCredentialsException();
-			HtmlPage page3 = page2.getAnchorByText("Homeworks").click();
-			final HtmlTable $ = (HtmlTable) page3.getByXPath("//table[@class='stud-hw-table']").get(0);
+			final HtmlTable $ = (HtmlTable) ((HtmlTable) page2.getAnchorByText("Homeworks").click()).getByXPath("//table[@class='stud-hw-table']").get(0);
```

Comment:
Does this really compile for you?
This warning was there because the Spartanizer tip that changes this line didn't compile for me.

Reply:
Note that there is 2 castings, instead of one, so it works :)

Reply:
Ohhhh so that was the problem, thanks for solving that!